### ISSUE TITLE
训练一开始loss为负值，且随着训练变大

### ISSUE THREAD

**Issue by zjf1075281201** [Sierkinhane/crnn_chinese_characters_rec#88]
作者你好，请问一下我在训练的时候训练一开始loss为负值，训练一段时间后loss反而变大了，但是绝对值一直是小于1的，调整了很多次学习率都是类似的情况，有人遇到过吗
下面是训练开始的部分
[0/10][5/204976] Loss: -0.127944
[0/10][10/204976] Loss: 0.175103
[0/10][15/204976] Loss: 0.192796
[0/10][20/204976] Loss: 0.220397
[0/10][25/204976] Loss: 0.240204
[0/10][30/204976] Loss: 0.256099
[0/10][35/204976] Loss: 0.270218
[0/10][40/204976] Loss: 0.280828
[0/10][45/204976] Loss: 0.292282
[0/10][50/204976] Loss: 0.298902
[0/10][55/204976] Loss: 0.307545

**Comment by zjf1075281201** [comment]
已基本解决，感谢作者

**Comment by jokaon** [comment]
我也遇到了这种情况，请问一下，你是如何解决的？非常感谢． @zjf1075281201

**Comment by leyiwang** [comment]
<img width="422" alt="image" src="https://user-images.githubusercontent.com/10727709/56977326-7cdbc980-6ba7-11e9-9ea6-65fc662aa236.png">
你好，我在训练过程中也遇到类似问题，起始loss为负，且逐渐增大。请问这是什么原因造成的，该怎么解决？@zjf1075281201, @haneSier

**Comment by monologue1107** [comment]
> 已基本解决，感谢作者

同问，可以知道你是如何解决的吗？非常感谢。QQ602729164

**Comment by leyiwang** [comment]
使用wrap_ctc的CTCLoss。首先安装wrap_ctc，取消第10行、160行注释，然后注释crnn_main.py模块的161行，即可。
```
line 10:  from warpctc_pytorch import CTCLoss
line 160: criterion = CTCLoss()
line 161:  # criterion = torch.nn.CTCLoss()
```
@monologue1107 @jokaon

**Comment by Sierkinhane** [comment]
@monologue1107 @jokaon @leyiwang pytorch的ctcloss有点问题，现在已经修复
